src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +251,230 @@ def inner(tree, path, root_path):
+    ------
+    ValueConflict
+        If both ``wildcard`` and ``match_whole_word`` are ``True``.
+
+    Examples
+    --------
+    >>> import ansys.fluent.core as pyfluent
+    >>> pyfluent.search("font", match_whole_word=True)
+    >>> pyfluent.search("Font")
+    >>> pyfluent.search("iter*", wildcard=True)
+    >>> pyfluent.search("读", language="cmn")   # search 'read' in Chinese
+    The most similar API objects are:
+    <solver_session>.file.read (Command)
+    <solver_session>.file.import_.read (Command)
+    <solver_session>.mesh.surface_mesh.read (Command)
+    <solver_session>.tui.display.display_states.read (Command)
+    <meshing_session>.tui.display.display_states.read (Command)
+    """
+    if wildcard and match_whole_word:
+        raise ValueConflict()
+    elif language and match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will match exact string and will turn off semantic matching.",
+            UserWarning,
+        )
+    elif match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will turn off semantic and wildcard matching.",
+            UserWarning,
+        )
+
+    api_object_names = get_api_tree_file_name(name=True)
+    names_txt = open(api_object_names, "r")
+    names = [name.replace("\n", "") for name in names_txt.readlines()]
+    synset_1 = wn.synsets(search_string, lang=language)
+
+    if wildcard and not match_whole_word:
+        queries = _process_wildcards(search_string, names)
+    elif synset_1 and not match_whole_word and not wildcard:
+        similar_keys = set()
+        for name in names:
+            synset_2 = wn.synsets(name, lang="eng")
+            for s1 in synset_1:
+                for s2 in synset_2:
+                    name_s1 = s1.name().split(".")[0]
+                    name_s2 = s2.name().split(".")[0]
+                    if search_string in name_s2 or name_s1 in name_s2:
+                        similar_keys.add(name_s2 + "*")
+        queries = set()
+        for key in similar_keys:
+            queries.update(set(_process_wildcards(key, names)))
+        queries = list(queries)
+        wildcard = True
+    elif match_whole_word and not wildcard:
+        queries = _process_misspelled(
+            word=search_string,
+            names_txt_file=api_object_names,
+            names=names,
+            match_whole_word=match_whole_word,
+        )
+    elif match_case:
+        queries = _process_misspelled(
+            word=search_string,
+            names_txt_file=api_object_names,
+            names=names,
+            match_case=match_case,
+        )
+
+    api_tree = get_api_tree_file_name(text=True)
+    api_tui_tree = get_api_tree_file_name(tui=True)
+    text_files = [api_tree, api_tui_tree]
+    print("\n The most similar API objects are: \n")
+    for text_file in text_files:
+        text = open(text_file, "r")
+        api_objects = [api_object.replace("\n", "") for api_object in text.readlines()]
```

Comment:
This can be rewritten in the same way as the similar code.

Reply:
Done. Thank you.

Reply:
@seanpearsonuk 

We are getting following error after updating this change.

`[line.rstrip('\n') for line in open(api_object_names, "r")]`

![image](https://github.com/ansys/pyfluent/assets/106588300/809431cf-775e-4831-beb4-31f146552a8e)

Earlier some of them were passing for following line. (others were failing due to NLTK data packages)

`[api_object.replace("\n", "") for api_object in text.readlines()]`

![image](https://github.com/ansys/pyfluent/assets/106588300/1f8ca752-d595-4f15-92b9-020c1ea5daca)